runtime/lua/vim/diagnostic.lua

```diff
@@ -60,7 +60,24 @@ local bufnr_and_namespace_cacher_mt = {
 }
 
 local diagnostic_cleanup = setmetatable({}, bufnr_and_namespace_cacher_mt)
-local diagnostic_cache = setmetatable({}, bufnr_and_namespace_cacher_mt)
+local diagnostic_cache = setmetatable({}, {
+  __index = function(t, bufnr)
+    if not bufnr or bufnr == 0 then
+      bufnr = vim.api.nvim_get_current_buf()
+    end
+
+    if rawget(t, bufnr) == nil then
```

Comment:
I don’t think the `rawget` is necessary as `__index` is only called if the key does not exist.